WalkingChallenge/SettingsViewController.swift

```diff
@@ -0,0 +1,205 @@
+  case accountSettings
+  case teams
+  case helpAndSupport
+}
+
+struct SettingsInfo {
+  let title: String
+  let values: [SettingsnValue]
+}
+
+struct SettingsnValue {
+  let configValue: String
+}
+
+class SettingsSectionDataSource {
+
+  let pedometerSource = UserInfo.pedometerSource
+
+  // swiftlint:disable line_length
+  var settingsInfo: [SettingsInfo] = [
+    SettingsInfo(title: Strings.Settings.accountSettings,
+                  values: [SettingsnValue.init(configValue: Strings.Settings.editProfile),SettingsnValue.init(configValue: Strings.Settings.changeEmailAddress),SettingsnValue.init(configValue: Strings.Settings.notificationsAndReminders)]),
+    SettingsInfo(title: Strings.Settings.teams, values: [SettingsnValue.init(configValue: Strings.Settings.changeTeams)]),
+    SettingsInfo(title: Strings.Settings.helpAndSupport, values: [SettingsnValue.init(configValue: Strings.Settings.faq),SettingsnValue.init(configValue: Strings.Settings.contactUs)])
+    ]
+  // swiftlint:enable Line Length
+}
+
+fileprivate class SettingsDataSource: NSObject, UITableViewDataSource, UITableViewDelegate {
+
+  internal var dataSource: SettingsSectionDataSource?
+
+  override init() {
+    super.init()
+    self.dataSource = SettingsSectionDataSource()
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    if section == TableSection.device.rawValue {
+      return 1
+    } else {
+      let configuration = dataSource?.settingsInfo[section - 1]
+      return (configuration?.values.count)!
+    }
+  }
+
+  func numberOfSections(in tableView: UITableView) -> Int {
+    return (dataSource?.settingsInfo.count)!+1
+  }
+
+  func tableView(_ tableView: UITableView,
+                 cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    var cell: UITableViewCell
+
+    switch indexPath.section {
+    case TableSection.device.rawValue:
+      // swiftlint:disable force_cast
+      let pedometerSource = dataSource?.pedometerSource
+      cell = tableView.dequeueReusableCell(withIdentifier: PedometerCell.identifier, for: indexPath)
+      (cell as! PedometerCell).deviceName.text = pedometerSource?.rawValue
+      (cell as! PedometerCell).deviceDescription.text = pedometerSource?.rawValue
+      (cell as! PedometerCell).devicePicture.backgroundColor = UIColor.brown
+      // swiftlint:enable force_cast
+      break
+    default:
+      let configure = dataSource?.settingsInfo[safe: indexPath.section - 1]
+      let justValue = configure?.values[indexPath.row]
+      cell = tableView.dequeueReusableCell(withIdentifier: "SettingsCell", for: indexPath)
+      cell.textLabel?.text = justValue?.configValue
+      break
+    }
+    cell.accessoryType = UITableViewCellAccessoryType.disclosureIndicator
+    return cell
+  }
+  func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    if section == TableSection.device.rawValue {
+      return Strings.Settings.device
+    } else {
```

Comment:
It's not always the if condition will be satisfied, hence else statement is needed here.